src/views/List.jsx

```diff
@@ -1,18 +1,44 @@
+import { useState, useMemo } from 'react';
+
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [itemSearch, setItemSearch] = useState('');
+
+	const searchedData = useMemo(() => {
+		if (!itemSearch) {
+			return data;
+		} else {
+			return data.filter((item) => {
+				return item.name.toLowerCase().includes(itemSearch);
+			});
+		}
+	}, [data, itemSearch]);
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
+			<form>
+				<label htmlFor="itemSearch">Search your shopping list:</label>
+				<input
+					type="text"
+					id="itemSearch"
+					name="itemSearch"
+					value={itemSearch}
+					onChange={(e) => setItemSearch(e.target.value.toLowerCase())}
```

Comment:
Great insight in setting the search value to lowercase! This is another place where you could use the `trim()` method if you wanted!

Reply:
Good idea. I will add `.trim()`  to the search input.